lib/evss/pciu/email_address.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module EVSS
+  module PCIU
+    class EmailAddress < BaseModel
+      # This regex comes from the `validates_format_of` Rails docs
+      #
+      # @see https://apidock.com/rails/ActiveModel/Validations/HelperMethods/validates_format_of
+      #
+      VALID_EMAIL_REGEX = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
```

Comment:
As per the Slack discussion, this will reject certain email addresses that are _technically_ valid, although uncommon. I don't have a strong opinion on how we should handle this, but it's probably worth having a strong consensus with the FE so we know what we're rejecting where.

Reply:
Can you give some examples of valid emails that will be rejected?

Reply:
There is email validation in the MHV email/secure messaging preferences, I believe in the controllers - would be good to check those and try to be consistent, although I understand if that's not possible due to differences in PCIU vs. MHV.

Reply:
@emersonthis - This regex ensures that:

- there is an `@` symbol
- there is something before the @ symbol
- there is a `.` after the @ symbol

I have found that these are the most common typos user’s make.

Based on our conversation in [this Slack thread](https://dsva.slack.com/archives/C909ZG2BB/p1522773434000788), I was under the impression that we had a consensus, but I could be wrong?

Here are the [specs for valid/invalid emails](https://github.com/department-of-veterans-affairs/vets-api/blob/e50fb94cb607d406a3c395d199b707aa390b8128/spec/lib/evss/pciu/email_address_spec.rb#L15-L31).  Can you pls let me know what other email examples that might be rejected, that we can add to our specs?

Reply:
`app/models/messaging_preferences.rb' and `prescription_preferences.rb` use: `/.+@.+\..+/i`

Reply:
@patrickvinograd - All the specs pass with that regex, as well.  

I'll use that one, for consistency, and set the ones in`app/models/prescription_preference.rb` and `app/models/messaging_preference.rb` equal to this constant, to act as a single source of truth.

I feel that when someone looks for any email related validation, this `email_address.rb` class will be at the top of their search results, and will be clear as to how we treat email validation.

Or would you recommend capturing, storing, and applying this one regex differently?

Thanks!

cc: @bshyong @ncksllvn

Reply:
Resolved in https://github.com/department-of-veterans-affairs/vets-api/pull/1815/commits/d5f13d5013811f20945904079f65d4b5346b50ff